tests/phpunit/tests/html-api/wpHtmlTagProcessor.php

```diff
@@ -601,6 +601,24 @@ public function test_next_tag_should_return_false_for_a_non_existing_tag() {
 		$this->assertFalse( $processor->next_tag( 'p' ), 'Querying a non-existing tag did not return false' );
 	}
 
+	/**
+	 * @ticket 61545
+	 */
+	public function test_next_tag_should_not_match_on_substrings_of_a_requested_tag() {
+		$processor = new WP_HTML_Tag_Processor( '<p><pic><picture>' );
+
+		$this->assertTrue(
+			$processor->next_tag( 'PICTURE' ),
+			'Failed to find a tag when requested: check test setup.'
+		);
+
+		$this->assertSame(
+			'PICTURE',
+			$processor->get_tag(),
+			'Should have skipped past substring tag matches, directly finding the PICTURE element.'
+		);
+	}
```

Comment:
In the case we had, this would have failed:

```php
$processor = new WP_HTML_Tag_Processor( '<p><img></p>' );
$this->assertFalse(
	$processor->next_tag( 'PICTURE' ),
	'Unexpectedly found a tag when requested: check test setup.'
);
```

Maybe good to have a negative case added to this as well.

Reply:
@westonruter I think these are the same case. in both cases they stop at the `P` - am I missing something?

Reply:
nevermind - I didn't understand your comment at first.

Reply:
added the test, but I used a different failure message